avm/res/net-app/net-app-account/main.bicep

```diff
@@ -33,6 +36,21 @@ param managedIdentities managedIdentitiesType
 @description('Optional. Array of role assignments to create.')
 param roleAssignments roleAssignmentType
 
+@description('Optional. The key name to use for encryption.')
```

Comment:
Hey @bobmclane999 thanks for this contribution. 
Please be aware in general that when it comes to CMK encryption we should leverage a common interface as detailed by AVM specs: https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#customer-managed-keys

In addition, a few months ago the implementation was not completed due to the NetApp encryption feature not being GA and additional requirement to enable a prerequisite at subscription scope. Ref CARML issue https://github.com/Azure/ResourceModules/issues/2991
cc @AlexanderSehr, @ahmadabdalla and last but not least the module owner @fbinotto to confirm

Reply:
Hi @eriqua  no problem this is my 1st attempt or a PR to you guys after a great Teams chat the other day with @jtracey93 so please be gentle with me :)

Have successfully managed to deploy with CMK and according to the below link it appears to be GA also have registered on our Non Prod subscription where I've been testing the code and appears to have registered so fingers crossed  - 
![image](https://github.com/Azure/bicep-registry-modules/assets/142896605/d6fee5a0-7af5-43c8-8070-5d5db1927ff0)
 
https://learn.microsoft.com/en-us/azure/azure-netapp-files/configure-customer-managed-keys?tabs=azure-portal

I see so on the CMK is that something you or Jack (@jtracey93 ) can help me with?

Kind Regards
Shaun @bobmclane999

Reply:
Hey @bobmclane999,

Firstly, thanks for the PR here. Ive just had a look and I think @eriqua  is saying we should use the AVM spec and use the User Defined Type and standard shared interface as we mandate in AVM resource modules for consumer consistency.

I have put in a PR (https://github.com/bobmclane999/bicep-registry-modules/pull/1) to your PR branch on your fork to do this, so you can see the changes, if you are happy please approve and merge them into your branch, which will then update this PR.

Could you please test it once you have merged my changes?

Hope this helps, let us know

Thanks 

Jack

Reply:
Oh and this feature looks to be GA, i concur

Reply:
Thanks @jtracey93 for chiming in and confirming!

@bobmclane999 regarding the screenshot you attached above, does the feature require to be registered or is that done by default? 
If the former @jtracey93 before merging, we should make sure the BRM target sub has that enabled. Also possibly we should add that prerequisite info to the README Notes section

Reply:
Unresolving due to latest comment

Reply:
Just checked that the feature is registered so we should be good to go from an environment prerequisites point of view. Resolving.